packages/poi-preset-karma/index.js

```diff
@@ -76,6 +71,36 @@ module.exports = (options = {}) => {
 
       delete webpackConfig.entry
 
+      if (coverage) {
+        /* for general usage */
+        webpackConfig.module.rules = [
+          {
+            test: /\.(jsx?)$/,
+            exclude: /(node_modules|\.test\.jsx?)/,
+            enforce: 'post',
+            loader: 'istanbul-instrumenter-loader',
+            query: {
+              esModules: true
+            }
+          }
+        ].concat(webpackConfig.module.rules)
+        /* for vue (assumes vue-loader) */
+        webpackConfig.module.rules = webpackConfig.module.rules
+          .map(r => {
+            const vueLoaderPos = r.use && r.use.findIndex(u => u.loader === 'vue-loader')
+            if (typeof vueLoaderPos === 'undefined' || vueLoaderPos === -1) {
+              return r
+            }
+            const options = r.use[vueLoaderPos].options
+            const instrumenterLoader = 'istanbul-instrumenter-loader?esModules=true'
+            options.preLoaders = (options.preLoaders || {})
+            options.preLoaders.js = typeof options.preLoaders.js === 'string' ?
```

Comment:
why it's preLoader for .vue file but postLoader for .js file?

Reply:
Actually it should be pre for both. If you use post it will run the coverage after the `babel-plugin-add-module-exports` has run which adds something like this to the top:

<img width="604" alt="screen shot 2017-10-10 at 1 51 11 pm" src="https://user-images.githubusercontent.com/247048/31371345-d8944f6c-adc2-11e7-9ad0-42ae6d2d0951.png">

Do you want me to change both to pre (as they should be) or can you correct it when you add the webpack-chain stuff @egoist

Reply:
thanks I will webpack-chain-ify it later today